sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/NumberPatternShouldBeRegularTest.cs

```diff
@@ -46,5 +46,37 @@ public void NumberPatternShouldBeRegular_FromCSharp7()
                 new NumberPatternShouldBeRegular(),
                 ParseOptionsHelper.FromCSharp7);
         }
+
+        [DataTestMethod]
+        [TestCategory("Rule")]
+        [DataRow("1_1_1", "Group size of 1")]
+        [DataRow("123_12", "First group is bigger then the second")]
+        [DataRow("1234_123_123", "First group is bigger then the others")]
+        [DataRow("123_123_12_123", "Another group has a different size")]
+        [DataRow("1_123.123_1234", "Last decimal group is bigger")]
+        [DataRow(".123_123_1234", "No group before the dot")]
+        public void HasIrregularPattern(string numericToken, string message)
+        {
+            Assert.IsTrue(NumberPatternShouldBeRegular.HasIrregularPattern(numericToken), message);
+        }
+
+        [DataTestMethod]
+        [TestCategory("Rule")]
+        [DataRow(".123_123_123_1", "No group before the dot")]
+        [DataRow("123", "No group character")]
+        [DataRow("1_123_123UL", "With UL suffix")]
```

Comment:
also add with "LU" suffix as it's legal, also to test case insensitivity
```
1_123_123LU
1_123_123lU
1_123_123Lu
1_1234_1234lu
1_12_13ul
```
and with [floating point suffixes](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/floating-point-numeric-types) (the below numbers are from the MS docs)

```
1.0
3D
0.42e2
3_000.5F
134.45E-2f
3_000.5m
1.5E6m
2.1m
```